js/scenes/Home.js

```diff
@@ -32,10 +34,333 @@ export class Home extends Component {
     // messes with react's diffing logic
     this.onSettingsPress = this.onSettingsPress.bind(this);
     this.renderRow = this.renderRow.bind(this);
+    this.onPanEnd = this.onPanEnd.bind(this);
+    this.onScroll = this.onScroll.bind(this);
 
     this.dataSource = new ListView.DataSource({
       rowHasChanged: (row1, row2) => row1 !== row2,
     });
+
+    this.state = {
+      listY: new Animated.Value(0),
+    };
+
+    // create a clamped property that we can
+    // use for translating the list view
+    this.translateY = this.state.listY.interpolate({
+      inputRange: [0, Header.HEIGHT],
+      outputRange: [0, Header.HEIGHT],
+      extrapolate: 'clamp',
+    });
+
+    // fades the header in/out in sync with list position
+    this.opacity = this.translateY.interpolate({
+      inputRange: [0, Header.HEIGHT],
+      outputRange: [0, 1],
+    });
+
+    // scales the header in/out in sync with list position
+    this.scale = this.translateY.interpolate({
+      inputRange: [0, Header.HEIGHT],
+      outputRange: [0.95, 1],
+    });
+
+    // tracks view's Y value before offset applied
+    this.baseY = 0;
+
+    // tracks previous Y value before offset applied
+    this.previousBaseY = 0;
+
+    // tracks Y position of the list including the offset
+    this.offsetY = 0;
+
+    // tracks Y offset applied to the baseY
+    this.yOffset = 0;
+
+    // tracks final rendered Y position of the view
+    this.clampedY = 0;
+
+    // tracks if scrolling is enabled on the ListView
+    this.scrollEnabled = true;
+
+    // tracks current scroll position of the ListView
+    this.scrollY = 0;
+
+    this.state.listY.addListener((e) => {
+      const offsetY = e.value;
+
+      this.offsetY = offsetY;
+      this.clampedY = this.clamp(this.offsetY).value;
+
+      // adjusting the offset during animations introduces bugs
+      if (!this.isPanning) return;
+
+      // get any delta between the current offsetY
+      // and the clampedY so that we can correct it
+      const deltaY = offsetY - this.clampedY;
+
+      // apply the delta to the offset ensuring that when the
+      // user's finger drags beyond the clamped zone, the moment
+      // they drag it in the opposite direction, the view responds
+      this.setOffset(this.yOffset - deltaY);
+    });
+  }
+
+  componentWillMount() {
+    this._panResponder = PanResponder.create({
```

Comment:
In the POI Map PR you moved this to its own methods which I think makes it clearer.